packages/shared/src/components/modals/PostModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactElement, useContext, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { StyledModal, ModalProps } from './StyledModal';
+import { useHideOnModal } from '../../hooks/useHideOnModal';
+import { useResetScrollForResponsiveModal } from '../../hooks/useResetScrollForResponsiveModal';
+import { PostContent, PostContentProps } from '../post/PostContent';
+import styles from './PostModal.module.css';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+
+interface PostModalProps
+  extends ModalProps,
+    Pick<PostContentProps, 'navigation' | 'isFetchingNextPage'> {
+  id: string;
+}
+
+export function PostModal({
+  className,
+  children,
+  onRequestClose,
+  id,
+  navigation,
+  isFetchingNextPage,
+  ...props
+}: PostModalProps): ReactElement {
+  const [currentPage, setCurrentPage] = useState('');
+  const { trackEvent } = useContext(AnalyticsContext);
+  const isExtension = !!process.env.TARGET_BROWSER;
+  useResetScrollForResponsiveModal();
+  useHideOnModal(props.isOpen);
+
+  useEffect(() => {
+    if (isExtension) {
+      return;
+    }
+
+    if (!currentPage) {
+      setCurrentPage(window.location.pathname);
+    }
+
+    window.history.replaceState({}, `Post: ${id}`, `/posts/${id}`);
+  }, [id]);
+
+  const onClose: typeof onRequestClose = (e) => {
+    if (!isExtension) {
+      window.history.replaceState({}, `Feed`, currentPage);
+      setCurrentPage('');
+    }
+    onRequestClose(e);
+  };
+
+  useEffect(() => {
+    trackEvent({
+      event_name: 'article modal view',
+      target_id: id,
+    });
+  }, []);
```

Comment:
This should actually fire based on the `id` prop as we need this for the tracking.

Reply:
I agree, though we track the changes in the navigation part where we also include the id.

Reply:
Since there will be some changes in the tracking of navigation, I shall apply your comment here.

Reply:
I meant in the useEffect we should put the id as a dependency, that way we are sure this fires correct.

Reply:
Yes. That wasn't necessary initially since in the navigation we provide the new ids. Now that we changed the parameters for the navigation and uses the old id we now will apply the `id` as an actual dependency here.